src/pkgcheck/checks/codingstyle.py

```diff
@@ -903,3 +903,102 @@ def feed(self, pkg):
                         yield RedundantDodir(
                             cmd.group('cmd'), line=dodir.group('call'),
                             lineno=lineno - 1, pkg=pkg)
+
+
+class EbuildUnquotedVariable(results.LineResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class EclassUnquotedVariable(results.EclassResult, results.Style):
+    """Variable is unquoted, but should be quoted"""
+
+    def __init__(self, variable, lineno, **kwargs):
+        super().__init__(**kwargs)
+        self.variable = variable
+        self.lineno = lineno
+
+    @property
+    def desc(self):
+        return f"{self.lineno}: Variable {self.variable} should be quoted"
+
+
+class _UnquotedVariablesCheck(Check):
+    message_commands = (
+        "die", "echo", "eerror", "einfo", "elog", "eqawarn", "ewarn"
+    )
+    var_names = (
+        "D", "DISTDIR", "FILESDIR", "S", "T", "ROOT", "BROOT", "WORKDIR", "ED",
+        "EPREFIX", "EROOT", "SYSROOT", "ESYSROOT", "TMPDIR", "HOME",
+        # variables for multibuild.eclass
+        "BUILD_DIR",
+    )
+
+    def _var_needs_quotes(self, pkg, node):
+        pnode = node.parent
+        while pnode != node:
+            if pnode.type == 'string':
```

Comment:
All the if/elif statements that check `pnode.type` and return False should be reduced to a single check for set membership for a set created during class init. You can move all the related comments to where that set is created as well.

Reply:
Done.